Improve CISupport testing

gsnedders opened the pull request:
#### eafd98a20440c864b45f056a5eb71d6c77ef0d15
<pre>
Improve CISupport testing
<a href="https://bugs.webkit.org/show_bug.cgi?id=302418">https://bugs.webkit.org/show_bug.cgi?id=302418</a>

Reviewed by Jonathan Bedard.

* Tools/CISupport/ews-build/factories.py:
(ServicesFactory.__init__): Add RunSharedUnitTests.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Add RunSharedUnitTests.
* Tools/CISupport/ews-build/steps.py:
(RunBuildWebKitOrgUnitTests): Move to run-tests.
(RunEWSUnitTests): Move to run-tests.
(RunSharedUnitTests): Added.
* Tools/CISupport/ews-build/steps_unittest.py: Update for above.
* Tools/CISupport/run-tests: Tidy up output handling.
* Tools/CISupport/runUnittests.py:
(main): Make this just call run-tests.

Canonical link: <a href="https://commits.webkit.org/306347@main">https://commits.webkit.org/306347@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d71869253756054df85dccc87aa1a64f7be9848e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/141012 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/13396 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/2668 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/149514 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/94108 "Built successfully") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/142885 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/14106 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/13548 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/108268 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/78467 "Passed tests") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/143963 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/10885 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/126211 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/89178 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/10482 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/8071 "Passed tests") | [⏳ 🛠 wpe-libwebrtc ](https://ews-build.webkit.org/#/builders/WPE-LibWebRTC-Build-EWS "Waiting in queue, processing has not started yet") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/119733 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/167/builds/2198 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/151976 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/13082 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/2492 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/116432 "Passed tests") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/140435 "Passed tests") | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/13097 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/11436 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/116776 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/12826 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/122881 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/68255 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/21768 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/13125 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/170/builds/2354 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/12864 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/13063 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/12908 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

bls1999 commented on Tools/CISupport/ews-build/factories_unittest.py:
```diff
@@ -791,6 +791,7 @@ class TestExpectedBuildSteps(unittest.TestCase):
             'buildbot-check-config-for-build-webkit',
             'ews-unit-tests',
             'buildbot-check-config-for-ews',
+            'Shared-unit-tests',
```
Nit: This should be a lowercase S to conform to step naming conventions.

bls1999 commented on Tools/CISupport/runUnittests.py:
```diff
@@ -1,59 +1,75 @@
 
 
 def main():
+    print('runUnittests.py is deprecated!', file=sys.stderr)
```
There should be a note here to use `./run-tests` instead.

gsnedders replied:
It does output what you should run instead below, once it has actually processed the arguments. But maybe we could put it here too.

gsnedders committed:
Improve CISupport testing
https://bugs.webkit.org/show_bug.cgi?id=302418

Reviewed by Jonathan Bedard.

* Tools/CISupport/ews-build/factories.py:
(ServicesFactory.__init__): Add RunSharedUnitTests.
* Tools/CISupport/ews-build/factories_unittest.py:
(TestExpectedBuildSteps): Add RunSharedUnitTests.
* Tools/CISupport/ews-build/steps.py:
(RunBuildWebKitOrgUnitTests): Move to run-tests.
(RunEWSUnitTests): Move to run-tests.
(RunSharedUnitTests): Added.
* Tools/CISupport/ews-build/steps_unittest.py: Update for above.
* Tools/CISupport/run-tests: Tidy up output handling.
* Tools/CISupport/runUnittests.py:
(main): Make this just call run-tests.

Canonical link: https://commits.webkit.org/306347@main
